app/collection/[id]/page.tsx

```diff
@@ -2,23 +2,114 @@
-
-const CollectionPage = (props: any) => {
-  const pb = new PocketBase("https://grotto-beasts-test.fly.dev");
-  console.log(pb.authStore.token);
-
-  return (
-    <div>
-      <SearchBar></SearchBar>
-      <StickyBox className="deckSticky" offsetTop={20} offsetBottom={20}>
-        <Deck collectionView={true} />
-      </StickyBox>
-      <CardList collectionView={true}></CardList>
-    </div>
-  );
-};
-
-export default CollectionPage;
+import Deck from '../../components/deck';
+import SearchBar from "../../components/searchBar";
+import { SaveButton}  from '../../components/saveButton'
+import { useContext, useEffect, useState } from "react";
+import {
+    CardsData,
+    CardDataContext,
+} from "../../components/providers/cardDataProvider/CardDataProvider";
+import { PocketBaseContext } from "../../components/providers/pocketBaseProvider/PocketBaseProvider";
+import Swal from 'sweetalert2';
+
+type collectionCountObjType = { [key: number]: number };
+
+const setCollectionCounts = (collectionCounts: any, cardsData: any) => {
+    for (var collectionCount in collectionCounts) {
+        var currCard = cardsData[parseInt(collectionCount)];
+        if (currCard) currCard.collectionCount = collectionCounts[parseInt(collectionCount)];
+    }
+    return cardsData;
+}
+
+async function getCollectionCount(id: string, cardsData: any) {
+    const res = await fetch(
+        `https://grotto-beasts-test.fly.dev/api/collections/cardCollection/records/${id}`
+    );
+    const data = await res.json();
+    return data;
+}
+
+const createCollectionCountObj = (cardList: CardsData[]) => {
+    var collectionCountObj: collectionCountObjType = {};
+    for (var index in cardList) {
+        var card = cardList[index];
+        if (card.collectionCount > 0) {
+            collectionCountObj[parseInt(index)] = card.collectionCount;
+        }
+    }
+    return collectionCountObj;
+}
+
+async function saveCollectionCount(id: string, collectionCountObj: collectionCountObjType, userId: string) {
+    const patchData = {collection: collectionCountObj, user: userId};
+    const res = await fetch(
+        `https://grotto-beasts-test.fly.dev/api/collections/cardCollection/records/${id}`,
+        {
+            method: "PATCH",
+            cache: "no-cache",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify(patchData)
+        }
+    );
+    if (res.status === 200) {
+        Swal.fire({
+            title: '<strong>Collection Saved!</strong>',
+            icon: 'success',
+            confirmButtonColor: '#257d52',
+            confirmButtonText: 'OK'
+          });
+    }
+    const data = await res.json();
+    return data;
+}
+
+export default function CollectionPage({ params }: any) {
+    const cardsData = useContext(CardDataContext);
+    const pocketBaseConnection = useContext(PocketBaseContext);
+    const [cardList, setCardList] = useState(Object.assign({}, cardsData));
```

Comment:
I'm curious if there's a way to avoid duplicating the card list in memory. But this is just a minor nit-pick 😝